src/coreclr/jit/optimizer.cpp

```diff
@@ -6582,21 +6587,6 @@ bool Compiler::optVNIsLoopInvariant(ValueNum vn, unsigned lnum, VNToBoolMap* loo
             }
         }
     }
-    else
```

Comment:
You are removing this block of code.

Was this used prior to your changes?

Reply:
The then clause now handles that bit  of logic (via the `VNF_MemOpaque` addition).